### ISSUE TITLE
Multiple active clients

### ISSUE THREAD

**Issue by Casper-Olsen** [seblj/roslyn.nvim#260]
Hi

_I can't link/show the repo, since it's not public._

## Description
I have an issue where I have two Active Clients for roslyn.
This leads to the LSP showing incorrect diagnostic information.

Like this:
```
vim.lsp: Active Clients ~
- roslyn (id: 1)
  - Version: ? (no serverInfo.version response)
  - Root directory: nil
  - Command: { "/home/casper/.local/share/nvim/mason/bin/roslyn", "--logLevel=Information", "--extensionLogDirectory=/home/casper/.local/state/nvim", "--stdio" }
  - Settings: {}
  - Attached buffers: 3
- roslyn (id: 2)
  - Version: ? (no serverInfo.version response)
  - Root directory: ~/repos/dotnetproj
  - Command: { "/home/casper/.local/share/nvim/mason/bin/roslyn", "--logLevel=Information", "--extensionLogDirectory=/home/casper/.local/state/nvim", "--stdio" }
  - Settings: {}
  - Attached buffers: 6, 3
```

My roslyn config looks like this:
```
return {
  'seblyng/roslyn.nvim',
  ft = 'cs',
  config = function()
    require('roslyn').setup {
      filewatching = 'roslyn',

      lock_target = false,

      ignore_target = function(target)
        return string.match(target, '%.CI%.sln$') ~= nil
      end,
    }

    vim.keymap.set('n', '<leader>rr', '<cmd>Roslyn restart<CR>', { desc = '[R]oslyn [R]estart' })

    vim.keymap.set('n', '<leader>rt', '<cmd>Roslyn target<CR>', { desc = '[R]oslyn [T]arget' })
  end,
}
```

In the root directory I have two solutions.
The .normal .sln I use and an CI .sln.

Roslyn picks the normal .sln, because of the `ignore_target`. But this ends up with the two Active Clients.

If I remove the `ignore_target` function then I get the option to choose a .sln with :Roslyn target. Doing this i get one Active Client and the diagnostic information is correct.

Am I using the `ignore_target` incorrectly since I am getting two active Clients?
Or is this a bug?
